pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -576,9 +576,8 @@ in package-set { inherit pkgs lib callPackage; } self // {
           ];
           dontUnpack = false;
         } ''
-        unpackPhase
-        cd "''${sourceRoot:-.}"
-        patchPhase
+        # Unpack and patch
+        phases="''${prePhases[*]-} unpackPhase patchPhase" genericBuild
```

Comment:
To test this you would need to build a Haskell package while applying `haskellPackages.buildFromCabalSdist` to it.